### ISSUE TITLE
testNext broken with typescript@3.8.0-dev.20191105

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#4888]
Example build failure: https://circleci.com/gh/palantir/tslint/18322?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

```
The 'strict-boolean-expressions' rule threw an error in '/home/circleci/project/test/rules/strict-boolean-expressions/default/test.ts':
TypeError: Cannot read property 'length' of undefined
    at checkIdentifier (/home/circleci/project/node_modules/typescript/lib/typescript.js:51626:61)
    at checkExpressionWorker (/home/circleci/project/node_modules/typescript/lib/typescript.js:58332:28)
    at checkExpression (/home/circleci/project/node_modules/typescript/lib/typescript.js:58282:38)
    at getTypeOfExpression (/home/circleci/project/node_modules/typescript/lib/typescript.js:58258:58)
    at getRegularTypeOfExpression (/home/circleci/project/node_modules/typescript/lib/typescript.js:63839:48)
    at getTypeOfNode (/home/circleci/project/node_modules/typescript/lib/typescript.js:63750:24)
    at Object.getTypeAtLocation (/home/circleci/project/node_modules/typescript/lib/typescript.js:33354:31)
    at checkExpression (/home/circleci/project/build/src/rules/strictBooleanExpressionsRule.js:167:28)
    at cb (/home/circleci/project/build/src/rules/strictBooleanExpressionsRule.js:126:25)
    at visitNode (/home/circleci/project/node_modules/typescript/lib/typescript.js:17768:24)
test/rules/strict-boolean-expressions/default/test.ts.lint: Failed!
```

**Comment by piotrl** [comment]
We encounter the same for TypeScript 3.6.5 after we migrated to Angular 9.0.1.
* MacOs and Linux
* NodeJs 10.15 and 12.16

Different rule:

```
The 'no-redundant-jsdoc' rule threw an error in '/src/lib/dashboard.component.ts':
Error: Unexpected tag kind: JSDocAuthorTag

at checkTag (/node_modules/tslint/lib/rules/noRedundantJsdocRule.js:106:23)
at cb (/node_modules/tslint/lib/rules/noRedundantJsdocRule.js:62:25)
at Array.forEach (<anonymous>)
at walk (/node_modules/tslint/lib/rules/noRedundantJsdocRule.js:55:34)
at Rule.AbstractRule.applyWithFunction (/node_modules/tslint/lib/language/rule/abstractRule.js:39:9)
at Rule.apply (/node_modules/tslint/lib/rules/noRedundantJsdocRule.js:35:21)
at Linter.applyRule (/node_modules/tslint/lib/linter.js:214:29)
at /node_modules/tslint/lib/linter.js:155:85
at Object.flatMap (/node_modules/tslint/lib/utils.js:160:29)
at Linter.getAllFailures (/node_modules/tslint/lib/linter.js:155:32)
```

Affects also codelyzer:
https://github.com/mgechev/codelyzer/issues/955

**Comment by JoshuaKGoldberg** [comment]
Per #4914 and the roadmap in #4534, accepting PRs to fix this issue.

_(Apologies for removing the tag early - that was misleading and incorrect. This issue was left open intentionally.)_

**Comment by JoshuaKGoldberg** [comment]
This should have been fixed by #4915.

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
